spinta/types/denorm/link.py

```diff
@@ -0,0 +1,52 @@
+from spinta import commands
+from spinta.components import Context, Property, Model
+from spinta.exceptions import NoRefPropertyForDenormProperty, ReferencedPropertyNotFound
+from spinta.types.datatype import Denorm, Ref, Object
+from spinta.types.helpers import set_dtype_backend
+
+
+@commands.link.register(Context, Denorm)
+def link(context: Context, dtype: Denorm) -> None:
+    set_dtype_backend(dtype)
+
+    dtype.rel_prop = _get_denorm_prop(
+        dtype.prop.name,
+        dtype.prop,
+        dtype.prop.model
+    )
+
+
+def _get_denorm_prop(
+    name: str,
+    prop: Property,
+    model: Model,
+) -> Property:
+    models = model.manifest.models
+    name_parts = name.split('.', 1)
```

Comment:
I just realized, that support for denormalized data was implemented incorrectly:

https://github.com/atviriduomenys/spinta/blob/086fdf1c8a435ede2578d64b5d146679ef757346/spinta/manifests/tabular/helpers.py#L498-L500

So I think we need to fix how denormalized data are read first.

When reading property name that contains a dot in name, for example:

    a.b.c.d  type=string

Tabular reader should produce following metadata:

```python
{
    'type': None,
    'properties': {
        'b': {
            'type': None,
            'properties': {
                'c': {
                    'type': None,
                    'properties': {
                        'd': {
                            'type': 'string',
                        },
                    },
                },
            },
        },
    },
}
```

Then we need to merge this into `self.state.model.data['properties'][row['property']]`

Then we will have a proper flow of processing nested properties, instead of splitting dots. Because, there can be multiple situations, how these nested denormalized properties can work, they can point to other denormalized data, they can point to arrays, base model properties, etc.

So a better way to handle this complexity is to use visitor pattern, where we handle each nested property separately without knowing or caring about other nodes or its place in the structure.

Sorry, somehow I missed that in https://github.com/atviriduomenys/spinta/issues/216, but we still need to fix that, before going any further.